app/controllers/users/nbp_wallet_controller.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require 'rqrcode'
+
+module Users
+  class NbpWalletController < ApplicationController
+    skip_before_action :require_user!
```

Comment:
What if ... we would rather overwrite this method to check that the desired values (`saml_uid` and `omniauth_provider` = `nbp`) are already in the session? This would remove the custom `omniauth_provider_uid` added and remove duplicates (we are storing the same value in `omniauth_provider_uid` and `saml_uid`). 

This change would only require to move `session['omniauth_provider'] = omniauth_provider` in the `OmniauthCallbacksController` a bit up (so that it is done earlier).

Of course, with this change we would still be unable to access the `current_user` in this controller. However, it would ensure that the page is only accessed by legitimate users (the check is currently skipped completely, and only captured by the condition in `set_provider_details`).

Reply:
Good idea, done :)